test/lib/ufe/testUngroupCmd.py

```diff
@@ -0,0 +1,133 @@
+        standalone.uninitialize()
+
+    def setUp(self):
+        ''' Called initially to set up the Maya test environment '''
+        # Load plugins
+        self.assertTrue(self.pluginsLoaded)
+
+        # Open ballset.ma scene in testSamples
+        mayaUtils.openGroupBallsScene()
+
+        # Clear selection to start off
+        cmds.select(clear=True)
+
+    def testUngroupCommand(self):
+        '''Simple test for ungroup command.'''
+
+        mayaPathSegment = mayaUtils.createUfePathSegment(
+            "|transform1|proxyShape1")
+
+        usdSegmentBall5 = usdUtils.createUfePathSegment(
+            "/Ball_set/Props/Ball_5")
+        ball5Path = ufe.Path([mayaPathSegment, usdSegmentBall5])
+        ball5Item = ufe.Hierarchy.createItem(ball5Path)
+
+        usdSegmentBall3 = usdUtils.createUfePathSegment(
+            "/Ball_set/Props/Ball_3")
+        ball3Path = ufe.Path([mayaPathSegment, usdSegmentBall3])
+        ball3Item = ufe.Hierarchy.createItem(ball3Path)
+
+        usdSegmentProps = usdUtils.createUfePathSegment("/Ball_set/Props")
+        parentPath = ufe.Path([mayaPathSegment, usdSegmentProps])
+        parentItem = ufe.Hierarchy.createItem(parentPath)
+
+        parentHierarchy = ufe.Hierarchy.hierarchy(parentItem)
+        parentChildrenPre = parentHierarchy.children()
+        self.assertEqual(len(parentChildrenPre), 6)
+
+        newGroupName = ufe.PathComponent("newGroup")
+
+        # get the USD stage
+        stage = mayaUsd.ufe.getStage(str(mayaPathSegment))
+
+        # set the edit target to balls.usda
+        layer = stage.GetLayerStack()[1]
+        self.assertEqual("ballset.usda", layer.GetDisplayName())
+        stage.SetEditTarget(layer)
+
+        ufeSelectionList = ufe.Selection()
+        ufeSelectionList.append(ball5Item)
+        ufeSelectionList.append(ball3Item)
+
+        groupCmd = parentHierarchy.createGroupCmd(
+            ufeSelectionList, newGroupName)
+        groupCmd.execute()
+
+        # Group object (a.k.a parent) will be added to selection list. This behavior matches the native Maya group command.
+        globalSelection = ufe.GlobalSelection.get()
+
+        groupPath = ufe.Path([mayaPathSegment, usdUtils.createUfePathSegment("/Ball_set/Props/newGroup1")])
+        self.assertEqual(globalSelection.front(), ufe.Hierarchy.createItem(groupPath))
+
+        # check parentHierarchy children size
+        self.assertEqual(len(parentHierarchy.children()), 5)
+
+        # ungroup
+        groupHierarchy = ufe.Hierarchy.hierarchy(globalSelection.front())
+        ungroupCmd = groupHierarchy.ungroupCmd()
+        ungroupCmd.execute()
+
+        # check parentHierarchy children size
+        self.assertEqual(len(parentHierarchy.children()), 6)
+
+        ungroupCmd.undo()
+
+        # check parentHierarchy children size
+        self.assertEqual(len(parentHierarchy.children()), 5)
+
+        ungroupCmd.redo()
+
+        self.assertEqual(len(parentHierarchy.children()), 6)
+
```

Comment:
A simple test to show the basic ungroup operation. I will add a better one with more meat later.

Reply:
@ppt-adsk  regarding the unit tests, I was planning to add more cases to cover exactly what you mentioned in ^ comments but was sidetracked with other works. 

Also, I found a crash in outliner when trying to **undo multiple ungroups**. It looks like something to do with outliner draw and notifications. I haven't had a chance to look it in details but I have it in my TODO list for tomorrow.

I will have 2-3 more test cases here soon.